tests/test_extract.py

```diff
@@ -0,0 +1,118 @@
+                texts=["hello Roberta, how are you doing today?"])
+
+    def test_varying_amount_of_entities(self):
+        examples = [
+            Example(
+                text="the bananas are red",
+                entities=[Entity(type="fruit", value="bananas"), Entity(type="color", value="red")]),
+            Example(
+                text="i love the color blue",
+                entities=[Entity(type="color", value="blue")]),
+            Example(
+                text="i love apples",
+                entities=[Entity(type="fruit", value="apple")]),
+            Example(
+                text="purple is my favorite color",
+                entities=[Entity(type="color", value="purple")]),
+            Example(
+                text="wow, that apple is green?",
+                entities=[Entity(type="fruit", value="apple"), Entity(type="color", value="green")])]
+        texts = ["i love bananas", "my favorite color is yellow", "i love green apples"]
+
+        extractions = co.extract('medium', examples, texts)
+
+        self.assertIsInstance(extractions, Extractions)
+        self.assertIsInstance(extractions[0].text, str)
+        self.assertIsInstance(extractions[1].text, str)
+        self.assertIsInstance(extractions[2].text, str)
+        self.assertIsInstance(extractions[0].entities, list)
+        self.assertIsInstance(extractions[1].entities, list)
+        self.assertIsInstance(extractions[2].entities, list)
+        self.assertEqual(len(extractions[0].entities), 1)
+        self.assertEqual(len(extractions[1].entities), 1)
+        self.assertEqual(len(extractions[2].entities), 2)
+
+    def test_many_examples_and_multiple_texts(self):
+        examples = [
+            Example(
+                text="hello my name is John, and I like to play ping pong",
+                entities=[Entity(type="Name", value="John"), Entity(type="Game", value="ping pong")]),
+            Example(
+                text="greetings, I'm Roberta and I like to play golf",
+                entities=[Entity(type="Name", value="Roberta"), Entity(type="Game", value="golf")]),
+            Example(
+                text="let me introduce myself, my name is Tina and I like to play baseball",
+                entities=[Entity(type="Name", value="Tina"), Entity(type="Game", value="baseball")])]
+        texts = ["hi, my name is Charlie and I like to play basketball", "hello, I'm Olivia and I like to play soccer"]
+
+        extractions = co.extract('medium', examples, texts)
+
+        self.assertEqual(len(extractions), 2)
+        self.assertIsInstance(extractions, Extractions)
+        self.assertIsInstance(extractions[0].text, str)
+        self.assertIsInstance(extractions[1].text, str)
+        self.assertIsInstance(extractions[0].entities, list)
+        self.assertIsInstance(extractions[1].entities, list)
+        self.assertEqual(len(extractions[0].entities), 2)
+        self.assertEqual(len(extractions[1].entities), 2)
```

Comment:
Curious: should we (can we) make assertions on the extracted string values? e.g.

```python
self.assertEqual(extractions[0].text, "Charlie")
```

Do these tests run using the live Cohere API, or some mocked data?

Reply:
These tests are run against the Prod API, so they are more like end-to-end tests. I think in this case we can definitely do what you suggested without too much flakiness! I'll make the change

Reply:
This LGTM.
Not sure if we would like to add `docblocks` on the methods. 
For me, the methods are self-explanatory, but not sure for others.